src/pretix/plugins/banktransfer/views.py

```diff
@@ -886,3 +892,27 @@ def get_object(self, *args, **kwargs):
             organizer=self.request.organizer,
             pk=self.kwargs.get('id')
         )
+
+
+@method_decorator(xframe_options_exempt, 'dispatch')
+class SendInvoiceMailView(EventViewMixin, OrderDetailMixin, View):
+    def post(self, request, *args, **kwargs):
+        if not self.order:
+            raise Http404(_('Unknown order code or not authorized to access this order.'))
+        try:
+            validate_email(request.POST['email'])
+        except ValidationError:
+            messages.error(request, _('Please enter a valid email address.'))
+            return redirect(self.get_order_url())
+
+        last_payment = self.order.payments.last()
+        last_invoice = self.order.invoices.last()
+        if not last_payment or not last_invoice:
```

Comment:
Technically, we should probably validate that the payment is (a) a bank transfer and (b) in pending state. (And if we put this behind the setting, as suggested above, we'd also need to check that here.)

Reply:
How do you usually handle these error cases, which in theory should never be visible to users, because they only would be triggered if somebody manually called the endpoint despite the form being hidden? E.g., do you use translated strings for the error messages?

Reply:
We'd usually use a `messages.error` + redirect as well, in some cases a `Http404`/`PermissionDenied`.

On one hand, it's additional work translating a string for them, on the other hand, they are not really **never** shown to users. In this case, they might e.g. be shown if someone has an outdated tab open in their browser and then submits the form from there :)

Reply:
Good point, then I'll use translated strings there as well